### ISSUE TITLE
Input bam file

### ISSUE THREAD

**Issue by AmalMakrini** [kennyrebelo/NET_snrPeakFinder#1]
Dear, 
I hope this email finds you well. I am interested in using your script for peak detection in NETseq data. I have a question regarding the input files for the script.

I would like to know if the script accepts the single nucleotide BAM file generated by get_SNR_bam_merged_reads.py, or if it directly takes the BAM file obtained from Filter_internalPriming.py. It is not entirely clear to me from the documentation.

Thank you in advance for clarifying this matter. I appreciate your assistance.

Best regards,
Zine

**Comment by kennyrebelo** [comment]
Dear Zine,

I advise using the output .bam file from Filter_internalPriming.py as input for the NET_snrPeakFinder.py script.
The PeakFinder script should work using any of the .bam files you suggested as input, however, if you use as input a .bam file that has not been filtered for internal priming reads - some of the peaks you obtain may result from internal priming.

I hope this response was helpful and sorry for the unclear documentation. I will edit the documentation to make this point clear.

Kind regards,
Kenny

**Comment by AmalMakrini** [comment]
Dear;

I am writing to express my gratitude for your prompt response. However, I am still encountering an issue that I hope you could assist me with.

Upon filtering out my BAM file for internal priming, I observed that only 40503 reads remained out of the initial 4120217 reads. This significant reduction raises concerns, and I am uncertain if I might be overlooking something in the process.

For filtering, I utilized the following parameters: -s single -a TGG.. -g genome.fasta. The fastq files were initially sequenced in paired end, subsequently merged using bbmap.

I deeply appreciate your ongoing support and guidance on this matter.

Best regards,

Amal

**Comment by kennyrebelo** [comment]
Dear Amal,

It seems like you followed the correct instructions to filter reads resulting from internal priming.

However, there are a couple of things you can check or fine tune to make sure you are not being overly stringent during this filtering process.

The "-a" argument given to  [Filter_InternalPriming.py](https://github.com/kennyrebelo/Filtering_InternalPriming/blob/master/README.md) must match the sequence that is expected to be found in case of a internal priming event.

For example, in my previous analysis of NET-seq data, I trimmed the sequencing adaptors from both read pairs, reads were then merged (using bbmap) and subsequently aligned to the genome. At this point, I used the Filter_InternalPriming.py script, with "-a TGG.." as an argument since it matches the beginning of the adaptor sequence that is expected to follow the 3'-end of the merged reads. Please see Panel A:
![Figure A](https://ars.els-cdn.com/content/image/1-s2.0-S1046202318304845-gr2.jpg).

The 3'Adaptor (TGGAATTCTCGGGTGCCAAGG was the full 3' sequencing adaptor that I trimmed from the raw reads) would correspond to the sequencing adaptor that follows the 3'end of the merged read pair (Pol II pause site), and is therefore the sequence that is expected to be found in case of internal priming.

In summary, confirm that the sequence you are using in the "-a " argument matches the sequence you expect to find in internal priming cases. If you are using the correct sequence, you can alternatively decrease stringency of hits for internal priming by increasing the length or removing the wildcards given to the "-a" argument. Example, using "-a TGGAAT" instead of "-a TGG..".

I hope this helps!

Kind regards,
Kenny

**Comment by AmalMakrini** [comment]
Dear Kenny,

I sincerely appreciate your prompt and comprehensive response. Indeed, the data was generated using the same adapter as yours. I am currently in the process of implementing the Netseq pipeline and, to facilitate this, I am initially testing it on the public data published in your paper Prudêncio et al., 2022.

Upon being more stringent, I observed that the result remains unchanged, indicating that the parameter is indeed correct. The only difference lies in the fact that when I merged reads using BBMap, I did not utilize the "xloose" option, which affects the stringency of merging.

To provide some insight into my process, I first cleaned the data using Cutadapt in paired-end mode with the following parameters: "-a TGGAATTCTCGGGTGCCAAGG -A GATCGTCGGACTGTAGAACTCTGAAC -m 10 -e 0.05 --match-read-wildcards -n 1". Subsequently, I merged the reads using STAR and retain only uniquely mapped reads using the "-q 255" filter with Samtools. Following this, I filter out internal priming utilizing the command "Filter_InternalPriming.py -f merged_reads_uniquely_mapped_sortedBycoordinate.bam -s single -a TGGAA". Additionally, I utilize the same BAM file to generate SNR using "get_SNR_bam_merged_reads.py".

I also have a further question regarding Peakfinder. Can I use the entire genome as a BED file within the "-g" option, or would it be advisable to limit the search to gene coordinates to expedite script processing?

Once again, I extend my gratitude for your invaluable assistance.

Best regards,

Amal

**Comment by AmalMakrini** [comment]
Hi,

I am reaching out to you once again regarding the issue I've encountered with the Filter_internal_priming.py script.

It appears that I have identified the root cause of the problem or not. The script successfully identifies a large number of reads to retain. However, during the stage where it is supposed to extract them from the BAM file using fgrep, it misses a significant number of reads. The problem arises during the ID cleaning step in order to save reads without IP. This cleaning process unintentionally removes 2 last numbers from the IDs of the reads. Consequently, due to the altered numbers, redundancy is introduced into the IDs, significantly reducing the number of successfully extracted reads from the initial bam. Maybe i'm wrong 

readID=ID[0]**[:-2]**.strip(">")
readID=readID.strip("(")
readID=readID.split("/")[0] # parsing readID

Example: 
Initial read ID SRR12023937.37945237 become	SRR12023937.379452

Thank you in advance

Best regards,

**Comment by kennyrebelo** [comment]
Hi Amal,

Removing read IDs is definitely not an intended behavior. If you are seeing this, just like you said, it will lead to an over filtering of reads that do not have internal priming.

I will download these datasets and process them in order to see if I can reproduce this error.

Also, regarding your Peakfinder question, you should be able to run it on the entire genome. At max, it should take two or three hours to finish (from what I remember).

Thank you very much for sharing your input!

Best regards,
Kenny

**Comment by AmalMakrini** [comment]
Hi Kenny,

Thank you for your prompt response. I want to inform you that when I ran PeakFinder  on the cluster using the entire genome, it took approximately three days.

Thank you again

Best regards,

Amal

**Comment by kennyrebelo** [comment]
Hi Amal,

I will also confirm if that is a the usual time it takes to run the peakfinder script. I could be misremembering of how long it usually took for whole genome runs.

Thanks for letting me know. I will update you of the results once I have completed running the NET-seq pipeline.

Best regards,
Kenny
